pyiceberg/table/__init__.py

```diff
@@ -269,7 +269,7 @@ def _apply(self, updates: Tuple[TableUpdate, ...], requirements: Tuple[TableRequ
         existing_requirements = {type(requirement) for requirement in self._requirements}
         for new_requirement in requirements:
             if type(new_requirement) not in existing_requirements:
-                self._requirements = self._requirements + requirements
+                self._requirements = self._requirements + new_requirement
```

Comment:
```suggestion
                self._requirements = self._requirements + (new_requirement,)
```
CI error 
```
ERROR tests/integration/test_writes/test_writes.py::test_abort_table_transaction_on_exception[2] - TypeError: can only concatenate tuple (not "AssertRefSnapshotId") to tuple
```
https://github.com/apache/iceberg-python/actions/runs/13227984398/job/36921549224?pr=1631